contracts/utils/ExpressCallHandler.sol

```diff
@@ -10,27 +10,39 @@ import { IExpressCallHandler } from '../interfaces/IExpressCallHandler.sol';
  * token transfers and token transfers with contract calls between chains. Implements the IExpressCallHandler interface.
  */
 contract ExpressCallHandler is IExpressCallHandler {
-    // uint256(keccak256('prefix-express-give-token'));
-    uint256 internal constant PREFIX_EXPRESS_RECEIVE_TOKEN = 0x67c7b41c1cb0375e36084c4ec399d005168e83425fa471b9224f6115af865619;
+    // uint256(keccak256('prefix-express-receive-token'));
+    uint256 internal constant PREFIX_EXPRESS_RECEIVE_TOKEN = 0x88626ceac99c3f34cc516d2581131c4cfb1abd891462d045cfef258326d5b00c;
 
     /**
      * @notice Calculates the unique slot for a given express token transfer.
-     * @param payload the payload of the receive token
      * @param commandId The unique hash for this token transfer
+     * @param payloadHash The payload keccak for the receive token
      * @return slot The calculated slot for this token transfer
      */
-    function _getExpressReceiveTokenSlot(bytes calldata payload, bytes32 commandId) internal pure returns (uint256 slot) {
-        slot = uint256(keccak256(abi.encode(PREFIX_EXPRESS_RECEIVE_TOKEN, payload, commandId)));
+    function _getExpressCallerSlot(bytes32 commandId, bytes32 payloadHash) internal pure returns (uint256 slot) {
```

Comment:
I think this should at least include sourceChain, otherwise you can make two GMP calls from chains A and B with same payload, express execute using commandId for call A but source chain B, which will emit an express executed event including source chain B, and when GMP call from A arrives it'll still go through but source chain will actually be A. It's a weird behaviour, so I prefer to avoid skipping any of the GMP params here similar to express executor tracker.